cleverhans/future/tf2/utils_tf.py

```diff
@@ -31,3 +31,80 @@ def clip_eta(eta, norm, eps):
     factor = tf.minimum(1., tf.math.divide(eps, norm))
     eta = eta * factor
   return eta
+
+
+def random_exponential(shape, rate=1.0, dtype=tf.float32, seed=None):
+  """
+  Helper function to sample from the exponential distribution, which is not
+  included in core TensorFlow.
+  
+  shape: shape of the sampled tensor.
+  :rate: (optional) rate parameter of the exponential distribution, defaults to 1.0.
+  :dtype: (optional) data type of the sempled tensor, defaults to tf.float32.
+  :seed: (optional) custom seed to be used for sampling.
+  """
+  return tf.random.gamma(shape, alpha=1, beta=1. / rate, dtype=dtype, seed=seed)
+
+
+def random_laplace(shape, loc=0.0, scale=1.0, dtype=tf.float32, seed=None):
```

Comment:
I've fixed the implementation of the Laplace distribution: it should be the difference of two iid. random exponential vectors such that `rate = 1/scale`, shifted by the mean `loc`

Reply:
The original code is from here:
https://github.com/cleverhans-lab/cleverhans/blob/8c095c9a65b4d2dacffff6676a565ee5ec269875/cleverhans/utils_tf.py#L451-L458

Since the most popular attacks are on L-inf norm, I did not check this function's the correctness on L-1 and L-2 norm.

But I think that you're right.